topic-operator/src/test/java/io/strimzi/operator/topic/TopicOperatorConfigTest.java

```diff
@@ -271,4 +271,86 @@ void shouldMaskSecuritySensitiveConfigsInToString() {
         assertFalse(config.toString().contains("FORBIDDEN"));
     }
 
+    @Test
+    public void shouldConfigureCustomSasl() {
+        // given
+        var config = TopicOperatorConfig.buildFromMap(Map.of(
+              TopicOperatorConfig.BOOTSTRAP_SERVERS.key(), "localhost:1234",
+              TopicOperatorConfig.NAMESPACE.key(), "some-namespace",
+              TopicOperatorConfig.SASL_ENABLED.key(), "true",
+              TopicOperatorConfig.SASL_CUSTOM_CONFIG_JSON.key(), """
+                    {
+                        "sasl.mechanism": "WAS_SMK_IAM",
+                        "sasl.jaas.config": "some.custom.auth.iam.IAMLoginModule required;",
+                        "sasl.client.callback.handler.class": "some.other.nonstandard.iam.IAMClientCallbackHandler"
+                    }
+                    """
+        ));
+
+        // then
+        assertEquals("some-namespace", config.namespace());
+
+        var adminConfig = config.adminClientConfig();
+        // client.id is random, so check it's there then remove for an easier assertion on the rest of the map
+        assertTrue(!adminConfig.get("client.id").toString().isEmpty());
+        adminConfig.remove("client.id");
```

Comment:
You simply set a custom `STRIMZI_CLIENT_ID` in the given config and assert on that?